sql/src/main/scala/org/apache/sedona/sql/utils/Adapter.scala

```diff
@@ -166,8 +169,9 @@ object Adapter {
     if (fieldNames != null && fieldNames.nonEmpty) {
       val userData = "" + geom.getUserData.asInstanceOf[String]
       val fields = userData.split("\t")
-//      geom.setUserData(null) // Set to null will lead to the null pointer exception of the previous line. Not sure why.
-      (Seq(geom), fields)
```

Comment:
I guess while we are at it, we should also address this issue here as well (even though it's not as bad as a null pointer exception) -- also, see test case with pairrdd that should cover this code path